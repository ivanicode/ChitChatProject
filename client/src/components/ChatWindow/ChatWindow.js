import React from 'react';

export function ChatWindow () {
    return (
        <div className="chatWindow">
            <div className="conversation">
            
            </div>
            <textarea className="textArea" placeholder="Napisz wiadomość...">

            </textarea>
            <div className="sendMessageButtonDiv">
                <button className="sendMessageButton" type="button">
                wyślij
                </button>
            </div>
        </div>
    );   
}

export default ChatWindow;
