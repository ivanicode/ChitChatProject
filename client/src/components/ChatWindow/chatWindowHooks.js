import { useState } from "react"
import { io } from 'socket.io-client';
import { useFetch } from '../../common/hooks/useFetchHook';


export function useChatWindowHooks(){
    const {writeMessage, sendMessage, enter} = useSendMessage()

    return {
      writeMessage,
      sendMessage,
      enter
    }
}

export function useSendMessage(){ 
  const [messagesArray, setMessagesArray] = useState([])
  const [textMessage, setTextMessage] = useState('')
  const socket = io('http://localhost:8082', {
    transports: ['websocket']
  });
  
  function sendMessage(event){
    
    if(textMessage){
      socket.emit('chat message', [textMessage, document.cookie.slice(5)])
      const newMessageArray = [...messagesArray];
      newMessageArray.push(textMessage);
      setMessagesArray(newMessageArray);
      setTextMessage('');
    }

    socket.on('chat message', function(msg) {
      console.log('mg:', msg)
      
      
      //window.scrollTo(0, document.body.scrollHeight);
    });
    
    
  }
  function enter(event){
    if(event.key.toLowerCase() === 'enter'){
      sendMessage()
    }
  }
  function writeMessage(event){
    const message = event.target.value;
    setTextMessage(message)
  }
  /*socket.on('chat message', function(msg) {
      
    console.log("socket.on", msg)
  });*/
    return {writeMessage, sendMessage, messagesArray, enter}
    
}


