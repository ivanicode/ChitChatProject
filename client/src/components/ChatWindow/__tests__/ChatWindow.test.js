import React from 'react';
import {shallow} from 'enzyme';
import {ChatWindow} from '../ChatWindow';


describe('ChatWindow component', () => {
    it('should match snapshot', () => {
        const wrapper = shallow(<ChatWindow />)
        expect(wrapper).toMatchSnapshot()
    })
})