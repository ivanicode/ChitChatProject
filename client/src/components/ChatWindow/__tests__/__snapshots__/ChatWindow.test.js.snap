// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatWindow component should match snapshot 1`] = `
<div
  className="chatWindow"
>
  <div
    className="conversationDiv"
  />
  <div
    className="createAndSendMessage"
  >
    <textarea
      className="textArea"
      id="messageBox"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Napisz wiadomość..."
    />
    <div
      className="sendMessageButtonDiv"
    >
      <button
        className="sendMessageButton"
        id="send"
        onClick={[Function]}
        type="button"
      >
        wyślij
      </button>
    </div>
  </div>
</div>
`;
